### PR TITLE
Bug fix/cinfra rta make data

### PR DESCRIPTION
**Pull request by mchacki**
### Scope & Purpose

*Fix RTA make_data execution and tests on Replication2.
1) When a server is shutdown in RTA, we now call the "removeServer" API on it. This is the Indicator for Replication2 to do full failover on followers.
2) Removed some assertions that are not true shortly after a server is permanently removed, but DBServers have not yet updated all logs.
3) Captured a valid throw, within a noexcept stack, the throw only happens on permanently removed Servers.
4) For simplicity and to be able to react to it, added --replicationVersion flag to scripts/unittests to select replication2 as default version, so we do not need to fiddle around with extraArgs anymore, adapted CircleCI config for it.
*

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Replication2/ReplicatedLog/LogLeader.cpp [hunk 1]
```diff
@@ -298,6 +298,7 @@ void replicated_log::LogLeader::executeAppendEntriesRequests(
                                            ResolvedPromiseSet> {
                           auto guarded = self->acquireMutex();
                           if (!guarded->_didResign) {
+                            // TODO: This can throw when we register the Callback
```
Comment (mchacki) [inline arangod/Replication2/ReplicatedLog/LogLeader.cpp:301]: Note to Reviewer: I created a Followup card for this TODO.
I think it is okay, but want to talk to the team about this next year.

File: js/client/modules/@arangodb/testutils/instance-manager.js [hunk 2]
```diff
@@ -622,6 +622,24 @@ class instanceManager {
...
+    // Make sure we remove the server
+    for (let i = 0; i < 10; ++i) {
+      const res = arango.POST_RAW("/_admin/cluster/removeServer", JSON.stringify(serverId));
+      if (res.code === 404 || res.code === 200) {
+        // Server is removed
+        return;
+      }
+      // Server could not be removed, give supervision some more time
+      // and then try again.
+      require("internal").wait(0.2);
```
Comment (dothebart) [inline js/client/modules/@arangodb/testutils/instance-manager.js:635]: please at least print a `.` so we see a loop is in progress.

**Review by dothebart — APPROVED** [review]
Please make the js loop heard to the user. else lgtm js part.

File: js/client/modules/@arangodb/testutils/instance-manager.js (hunk 2 above)
Comment (mchacki) [reply js/client/modules/@arangodb/testutils/instance-manager.js:636]: Done

**Comment by mchacki** [comment]
Red test is unrelated.
Diff between green run before is a JS print only.

**Pull request merged**